src/app/modules/startup/controller.nim

```diff
@@ -68,43 +70,59 @@ proc newController*(delegate: io_interface.AccessInterface,
   result.tmpSeedPhraseLength = 0
   result.tmpKeychainErrorOccurred = true
   result.tmpRecoverUsingSeedPhraseWhileLogin = false
+  result.tmpSelectedLoginAccountIsKeycardAccount = false
 
 # Forward declaration
 proc cleanTmpData*(self: Controller)
 
+proc disconnect*(self: Controller) =
+  for id in self.connectionIds:
+    self.events.disconnect(id)
+
 proc delete*(self: Controller) =
-  discard
+  self.disconnect()
 
 proc init*(self: Controller) =
-  self.events.on(SignalType.NodeLogin.event) do(e:Args):
+  var handlerId = self.events.onWithUUID(SignalType.NodeLogin.event) do(e:Args):
     let signal = NodeSignal(e)
     self.delegate.onNodeLogin(signal.event.error)
+  self.connectionIds.add(handlerId)
```

Comment:
@osmaczko no, let me give you some context... we have keycard service which is the main point for communication between the desktop app and the keycard lib. Also there is a shared keycard module which is meant to be used from many places in the app since it knows how to run and handle a lot of common flows. Also some of the main flows may run some of the flows from keycard shared module (as well there is a flow in shared module which runs another shared module flow, this will come with the next `authentication` flow) because of all that from one side and Nim limitations from the other side (cause we don't connect signal of an instance to a slot of another instance, but put all signals in the hash table and artificially simulating signal/slot mechanism) we're forced to manually take care of what is connected to what in order to avoid triggering slots of the main flow while for example shared flow is run from it and main flow needs to continue the execution once shared flow is completed from the point where it runs shared flow. Now all these may look confusing, but here is an example... 
- `main flow` is migrating a certain keypair to a keycard
- in some point A during that action you have to enter your pin/password in order to complete that migration
- then from that `main flow` we're running shared module and run `authentication flow` from it
- at this moment `main flow` is blocked at point A and waits for success/failor of `authentication flow` in order to complete its flow
- now cause `main flow` and `authentication flow` are both connected to signals sent from keycard service, we have somehow to disconnect `main flow` while `authentication flow` is running and connect them again once `authentication flow` is done. 

That's something we haven't had so far. Hope that this example helps you understand the point, at least it's more clear than the description above it. :)